searx/searx
# [fix] make autofocus configurable (#1984)

"Results page: having text cursor still in search field is annoying #1984"


## What does this PR do?

This patch allows the text area autofocus on the results page to be toggled either in the UI or in the searx server config.


## Why is this change important?

This allows users to configure searx to not focus the text area on the results page allowing for more easy keyboard navigation when viewing searx results.

## How to test this PR locally?

Run a local instance. Set the "Autofocus search field" setting to "Off". Perform a search and check the text area is not focused.

Also:

* test that default existing behaviour still works so that the search area on the results page is still focused when "Autofocus search field" is set to on
* test that the main searx page has its text field focused independent of this setting.

This setting's default value can also be tested by changing the `ui/autofocus` setting to `False` in the `searx/settings.yml` file.


## Related issues

Closes #1984
